### DISCUSSION TITLE
Using msal for python in a mobile app developed in python and Kivy. Is there any way to open the browser in app itself in Interactive flow

### DISCUSSION THREAD

**Discussion by lijithomas88** [Azure-Samples/ms-identity-python-webapp#112] [Q&A]
We are developing a simple mobileapp using Python and Kivy. We are using microsoft-authentication-library-for-python for authentication using Azure AD B2C.

Tried out below one and authentication is working fine. 
https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/dev/sample/interactive_sample.py

But for authentication, right now its opening a browser outside the mobile app. Is there any way to open the browser inside the mobile app itself.

**Comment by rayluo** [comment]
Glad to know that the MSAL Python library and its `acquire_token_interactive()` works out of the box on Kivy.

We did not explore how or whether it is possible to launch an embedded browser (i.e. "the browser inside the mobile app itself"). MSAL Python itself is cross-platform and hardly specifies what browser to use. Besides, the system browser is a common denominator among the platforms that Kivy supports. Perhaps you can dig into Kivy's docs to see whether it can launch embedded browser when running on mobile?

BTW, your follow-up questions on this topic shall go into MSAL Python's repo, not this webapp sample repo.
